packages/editor/src/components/SvgUploader.tsx

```diff
@@ -0,0 +1,94 @@
+import { FileUploader } from "react-drag-drop-files";
+import toast from "react-hot-toast";
+import { useRecoilState } from "recoil";
+import { diagramMetadataSelector, fileContentsSelector } from "../state/atoms";
+
+export default function SvgUploader() {
+  const setDomain = useRecoilState(fileContentsSelector("domain"))[1];
+  const setSubstance = useRecoilState(fileContentsSelector("substance"))[1];
+  const setStyle = useRecoilState(fileContentsSelector("style"))[1];
+  const [diagramMetadata, setDiagramMetadata] = useRecoilState(
+    diagramMetadataSelector
+  );
+
+  const handleChange = (svg: File) => {
+    const reader = new FileReader();
+    reader.readAsText(svg);
+    reader.onabort = () => console.log("file reading was aborted");
+    reader.onerror = () => console.log("file reading has failed");
+    reader.onload = () => {
+      const svgText = reader.result?.toString();
+      if (!svgText) {
+        return;
+      }
+
+      const parser = new DOMParser();
+      const xmlDoc = parser.parseFromString(svgText, "text/xml");
+      const styElem = xmlDoc.getElementsByTagName("sty");
+      const subElem = xmlDoc.getElementsByTagName("sub");
+      const dslElem = xmlDoc.getElementsByTagName("dsl");
+
+      setDiagramMetadata((metadata) => ({
+        ...metadata,
+        variation: diagramMetadata.variation,
```

Comment:
I think you'll need to read the variation from the SVG file instead of using the one from the current workspace.